### PR TITLE
Allow multiple year renewal for domains

### PR DESCRIPTION
**Pull request by ayebare**
Related to #
pau2Xa-5jV-p2

Ticket tracking
https://github.com/orgs/Automattic/projects/388/views/4?pane=issue&itemId=38558926

## Proposed Changes
Currently, we only allow multiple-year domain purchases for new domain registrations/purchases,
This patch opens up multiple-year domain registrations/purchases to existing domain renewals.

## Testing Instructions
Instructions can be found here 
D128208-code

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/checkout/src/components/wp-order-review-line-items.tsx [hunk 1]
```diff
@@ -252,7 +252,7 @@ function LineItemWrapper( {
 	const shouldShowVariantSelector =
 		onChangeSelection &&
 		! isWooMobile &&
-		! isRenewal &&
+		( ! isRenewal || ( isRenewal && product.is_domain_registration ) ) &&
```
Comment (sirbrillig) [inline client/my-sites/checkout/src/components/wp-order-review-line-items.tsx]: Minor, but the mix of negatives, OR, and AND make this a bit hard to understand. We could make this more readable by creating another variable. Something like:

```js
const isNonDomainRenewal = isRenewal && ! product.is_domain_registration;
const shouldShowVariantSelector =
  // ...
  ! isNonDomainRenewal &&
  // ...
```

An alternative which I think may be even better, although it's more code, is to make an IIFE:

```js
const shouldShowVariantSelector = (() => {
  if ( ! onChangeSelection ) {
    return false;
  }
  // ...
  if ( isRenewal && ! product.is_domain_registration ) {
    return false;
  }
  return true;
} )();
```

Comment (ayebare) [reply client/my-sites/checkout/src/components/wp-order-review-line-items.tsx]: Thanks, updated.

**Comment by sirbrillig** [comment]
I haven't tested this (I leave that to your team) but from a code perspective, this looks good to me. 

Note: there's a type error right now,

```
client/my-sites/checkout/src/components/wp-order-review-line-items.tsx(316,7): error TS2322: Type 'OnChangeItemVariant | undefined' is not assignable to type 'OnChangeItemVariant'. Type 'undefined' is not assignable to type 'OnChangeItemVariant'.
```

I think this is happening because TypeScript can't read the IIFE like it could the condition, so it doesn't know that line 316 will never happen if `onChangeSelection` is falsy. We can solve this by making the condition more explicit (even if it duplicates logic a bit).

Change line 313 to read the following that that should handle it:
```diff
- { areThereVariants && shouldShowVariantSelector && (
+ { areThereVariants && shouldShowVariantSelector && onChangeSelection && (
```

**Review by gmovr — APPROVED** [review]
Tested this manually, and it works well. I can see the new renewal date, and it's correct. Testing variations of multi-year domains + renewals (e.g. a 2 year domains + manual renewal for 4 years) also work.

**Pull request merged**